### PR TITLE
Fix logging statement with missing parameter

### LINKED ISSUE #108355: Logging statement with missing parameter
https://github.com/kubernetes/kubernetes/blob/5bafc8306a7b622e292746aec8a9e9212b755071/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go#L225

> I0225 16:19:15.424936 1416311 versioning.go:225] a memory allocator was provided but the encoder %!T(MISSING) doesn't implement the runtime.EncoderWithAllocator, using regular encoder.Encode method

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
`kubernetes/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go`
Line 225 has a logging statement with missing parameter, causing:

```bash
I0225 16:19:15.424936 1416311 versioning.go:225] a memory allocator was provided but the encoder %!T(MISSING) doesn't implement the runtime.EncoderWithAllocator, using regular encoder.Encode method
```

Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #108355

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
Local unit test is working for me, will wait for reviewers bf triggering retest again

**Review by yuvicc — APPROVED** [review]

File: staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go
Comment (MadhavJivrajani) [inline staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go:225]:
```suggestion
			klog.V(4).Infof("a memory allocator was provided but the encoder %T doesn't implement the runtime.EncoderWithAllocator, using regular encoder.Encode method", c.encoder.Identifier())
```
We want to provide identifier of the `encoder` that does not implement `runtime.EncoderWithAllocator`, which is `c.encoder`. The panics in the tests are due to the fact that `encoder` is always `nil` since it does not implement the `EncoderWithAllocator` interface.

**Comment by MadhavJivrajani** [comment]
/hold
For squashing commits and accordingly changing the commit msg (provided the fix works)

**Comment by ArangoGutierrez** [comment]
rebased, squashed and updated, thanks

File: staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go
Comment (p0lyn0mial) [inline staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go:225]: `c.encoder.Identifier()` returns a `string` whereas the function uses `%T`,either change `%T to %s` or `c.encoder.Identifier() to c.encoder`

Comment (ArangoGutierrez) [reply staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go]: Changed `%T` to `%s` , thanks for your suggestion and time

**Comment by MadhavJivrajani** [comment]
/hold cancel
/lgtm
/assign @caesarxuchao 
(for approval)

**Comment by p0lyn0mial** [comment]
/lgtm

thanks for the fix.

**Comment by ArangoGutierrez** [comment]
Friendly PING @caesarxuchao

**Comment by ArangoGutierrez** [comment]
👋🏽 @caesarxuchao
/woof

**Review by ionicc — APPROVED** [review]

**Comment by ArangoGutierrez** [comment]
> /hold cancel /lgtm /assign @caesarxuchao (for approval)

/assign smarterclayton

**Comment by ArangoGutierrez** [comment]
Good morning reviewers, just passing by to wish you all a nice Monday coffee
/woof

**Comment by dims** [comment]
this is a one-liner what adds a missing parameter, so LGTM

/approve
/lgtm

**Pull request merged**
